docs/items/datacomponents.md

```diff
@@ -0,0 +1,324 @@
+    private final PatchedDataComponentMap components;
+
+    // Overloads can be provided to supply the map itself
+    public ExampleHolder() {
+        this.data = 0;
+        this.components = new PatchedDataComponentMap(DataComponentMap.EMPTY);
+    }
+
+    @Override
+    public DataComponentMap getComponents() {
+        return this.components;
+    }
+
+    @Nullable
+    @Override
+    public <T> T set(DataComponentType<? super T> componentType, @Nullable T value) {
+        return this.components.set(componentType, value);
+    }
+
+    @Nullable
+    @Override
+    public <T> T remove(DataComponentType<? extends T> componentType) {
+        return this.components.remove(componentType);
+    }
+
+    @Override
+    public void applyComponents(DataComponentPatch patch) {
+        this.components.applyPatch(patch);
+    }
+
+    @Override
+    public void applyComponents(DataComponentMap components) {
+        this.components.setAll(p_330402_);
+    }
+
+    // Other methods
+}
+```
+
+### `DataComponentPatch` and Codecs
+
+To persist components to disk or send information across the network, the holder could send the entire `DataComponentMap`. However, this is generally a waste of information as any defaults will already be present wherever the data is sent to. So, instead, we use a `DataComponentPatch` to send the associated data. `DataComponentPatch`es only contain the patch information of the component map without any defaults. The patches are then applied to the prototype in the receiver's location.
+
+A `DataComponentPatch` can be created from a `PatchedDataComponentMap` via `#patch`. Likewise, `PatchedDataComponentMap#fromPatch` can construct a `PatchedDataComponentMap` given the prototype `DataComponentMap` and a `DataComponentPatch`.
+
+```java
+public class ExampleHolder implements MutableDataComponentHolder {
+
+    public static final Codec<ExampleHolder> CODEC = RecordCodecBuilder.create(instance ->
+        instance.group(
+            Codec.INT.fieldOf("data").forGetter(ExampleHolder::getData),
+            DataCopmonentPatch.CODEC.optionalFieldOf("components", DataComponentPatch.EMPTY).forGetter(holder -> holder.components.asPatch())
+        ).apply(instance, ExampleHolder::new)
+    );
+
+    public static final StreamCodec<RegistryFriendlyByteBuf, ExampleHolder> STREAM_CODEC = StreamCodec.composite(
+        ByteBufCodecs.INT, ExampleHolder::getData,
+        DataComponentPatch.STREAM_CODEC, holder -> holder.components.asPatch(),
+        ExampleHolder::new
+    );
+
+    // ...
+
+    public ExampleHolder(int data, DataComponentPatch patch) {
+        this.data = data;
+        this.components = PatchedDataComponentMap.fromPatch(
+            // The prototype map to apply to
+            DataComponentMap.EMPTY,
+            // The associated patches
+            patch
+        );
+    }
+
+    // ...
+}
+```
+
+When to send the holder across the network or persist to disk is an exercise left to the reader.
```

Comment:
This just sounds bad. This should be something along the lines of "Actually syncing and writing to disk must be done manually" and have a link to the networking section.